runtime/packages/core-modules/assembly/executive.ts

```diff
@@ -27,12 +27,13 @@ export namespace Executive{
     export function initialChecks(block: Block): void{
         let header = block.header;
         let n: CompactInt = header.number;
+        // check that parentHash is valid
+        const previousHash: CompactInt = new CompactInt(n.value - 1);
+        const blockHash = Storage.get(Utils.stringsToBytes(["system", "block_hash"], true).concat(previousHash.toU8a()));
```

Comment:
I think that we could add `System.getHashAtBlock(blockNumber)` function. This way the `executive` module shouldn't care about the keys that we use for storing block hashes. We should avid hardcoding "system", "block_hash" or other keys to more than 1 place.